### DISCUSSION TITLE
UnauthenticatedFromAddress error in Mailwizz using Postal

### DISCUSSION THREAD

**Discussion by bukunmilab** [postalserver/postal#3159] [Help with using Postal] [answered]
Hi,
I already verified my Mailwizz domain and Postal domain in Postal server and Postal seems to have no issues sending emails. 

But I noticed an issue with sending campaign using Postal web API in Mailwizz. I think it might be related to authorization. 

I checked campaign delivery logs and found this:
[UnauthenticatedFromAddress] The From address is not authorised to send mail from this server

I have already enabled the "Allow Sender Header" from Postal panel and I expect this to automatically allow all sending domains outside of postal to be able to send out emails without such error.

Note: I have no problem sending out emails successfully with campaign and transactional API in Mailwizz as an admin since I automatically passed authorization as my Mailwizz domain is DNS-verified in Postal panel or server.

What my problem is, my customers are not able to send out emails without the errors I mentioned as Postal requires that all sending domains should be added to my postal server (and it is impossible to add each customer's domain and verify their DNS records in Postal like that. We will have tons of customers).

How can I resolve this?

**Comment by willpower232** [comment]
Are you definitely using the API and not SMTP? It might be better to use SMTP and then you can look at the SMTP logs to see exactly what content is being sent by Mailwizz.

**Reply by bukunmilab** [reply]
I have tried the SMTP Postal also in Mailwizz before I switched to the API option.
Here is the SMTP error log from Mailwizz when my customer tried to send out emails from Mailwizz:

`Expected response code "250" but got code "530", with message "530 From/Sender name is not valid".`

**Reply by willpower232** [reply]
Did you add at least one domain to Postal and are using it as the from of the message?

**Reply by bukunmilab** [reply]
Yes the from domain is added and verified in Postal. But since customers want to send emails as their own domain which is verified in Mailwizz, there is an option to change the sender domain to their own sender domain as this is the only way that makes sense in customer users' perspective. 

By the way my Mailwizz is hosted in a different VPS. This is done to optimize server specs and resources and to allow Postal it's own independence for ease of management from its own VPS.

I have brands and companies as customers and they want the customizations of sending as their own email domain, and I would understand that makes sense.

I would expect the "Allow Sender Header"  option in Postal to resolve this but it isn't doing what I expected. 

Also my customers should be responsible for their own sending activities including their own name as in most cases you have no idea what your costomers are sending so again we cannot Force our sender email domain on this customers. 

There has to be a way for Postal to know or tell postal to whitelist other domains sending through my SMTP or API credentials from an external app like Mailwizz, Mautic, etc.

**Reply by willpower232** [reply, marked as answer]
That error message is happening because the domain you've added to Postal isn't in the from or sender line on the email so either you haven't set it correctly or you need to swap them around maybe.

You can use `postal logs smtp` to see what Mailwizz is sending to Postal and see what the problem is.

**Comment by bukunmilab** [comment]
It seems I was able to fix the issue with your suggestions. Thanks a bunch.

Mailwizz delivery server settings - Postal SMTP
<img width="798" alt="Screenshot_1" src="https://github.com/user-attachments/assets/3377f4ef-55b4-4a11-b5e3-17f8dcaf7f48">

Postal panel - real-time message processing.

<img width="761" alt="Screenshot_2" src="https://github.com/user-attachments/assets/b3aba1a7-b193-465d-9c59-cb84402fbd1a">


Now comes another issue, after about 94 messages send successfully with the greenlight, gmail, aol, and yahoo and other ESPs started hardfailing and softfailing my messages. What do you suggest in this case?

**Reply by willpower232** [reply]
You would have to use things like mxtoolbox.com and mail-tester.com to spot any problems with your messages or your choice of server provider.
